src/Aspire.Dashboard/Extensions/StringExtensions.cs

```diff
@@ -49,4 +32,30 @@ static bool IsValidHtmlIdCharacter(char c)
             return char.IsLetterOrDigit(c) || c == '_' || c == '-';
         }
     }
+
+    /// <summary>
+    /// Returns the two initial letters of the first and last words in the specified <paramref name="name"/>.
+    /// If only one word is present, a single initial is returned. If <paramref name="name"/> is null, empty or
+    /// white space only, <paramref name="defaultValue"/> is returned.
+    /// </summary>
+    [return: NotNullIfNotNull(nameof(defaultValue))]
+    public static string? GetInitials(this string name, string? defaultValue = default)
```

Comment:
This can be written to only allocate the returned string. This will hardly ever be called, so it's not a big deal, but all the same :)

```c#
[NotNullIfNotNull(nameof(defaultValue))]
public static string? GetInitials(string name, string? defaultValue = default)
{
    ReadOnlySpan<char> s = name.AsSpan().Trim();

    if (s.Length == 0)
        return defaultValue;

    int lastSpaceIndex = s.LastIndexOf(' ');

    if (lastSpaceIndex == -1)
        return s[0].ToString().ToUpperInvariant();

    // The name contained two or more words. Return the initials from the first and last.
    return $"{char.ToUpperInvariant(s[0])}{char.ToUpperInvariant(s[lastSpaceIndex + 1])}";
}
```

Reply:
Yeah I didn't spend much time trying to make that better, just went with what worked to focus on the UI. But yours works, so I'll take it (slightly modified for analyzer warnings)